### PR TITLE
🧰 [FRD-604] Fix GnosisSafe 'reject by user' case / Add `isEvmContract` util

### PR DESCRIPTION
**Pull request by iower**

### CODE REVIEW

File: src/utils/evm.ts [hunk 1]
```diff
@@ -21,3 +23,14 @@ export const detectNewTxFromAddress: (address: string, provider: ethers.provider
     provider.on(filter, onFound)
   })
 }
+
+export const isEvmContract = async (chainId: number, address: string) => {
+  if (!isEvmChain(chainId)) {
+    throw new Error(`Non-EVM chainId ${chainId}`)
+  }
+  const rpc = getNetworkById(chainId).rpc_url
```
Comment (nrjshka) [inline src/utils/evm.ts]:
```suggestion
  const { rpc_url: rpc }  = getNetworkById(chainId)
```

File: src/utils/evm.ts [hunk 2]
```diff
@@ -21,3 +23,14 @@ export const detectNewTxFromAddress: (address: string, provider: ethers.provider
...
+  const provider = new ethers.providers.JsonRpcProvider(rpc)
+  const addressCode = await provider.getCode(address)
+  const isContract = addressCode !== EVM_NON_CONTRACT_ADDRESS_CODE
+  return isContract
```
Comment (nrjshka) [inline src/utils/evm.ts]:
```suggestion
  return addressCode !== EVM_NON_CONTRACT_ADDRESS_CODE
```

**Review by nrjshka — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@viaprotocol/web3-wallets",
-  "version": "1.0.15",
+  "version": "1.0.16",
```
Comment (insulineru) [inline package.json]: Please, upd version only when task in `master` branch. Might be conflicts plus need release commit anyway

**Review by insulineru — APPROVED** [review]
Agree with @nrjshka comments, other looks good!

File: package.json (hunk 3 above)
Comment (iower) [reply package.json]: Ok, sorry. Removed.

**Pull request merged**